canonical/vanilla-framework
# Baseline border fix

## Done
On low pixel density devices a 1.5px border would computed as 1px.
If we use $input-border-thickness = 1.5px in the calculation and the user is on a low pixel density screen then the border is rendered as 1px, but we still eg. subtract 1.5px from the padding which will throw off the baseline grid alignment.
So we have to check if the users screen is capable of displaying 0.5px steps at runtime. This is only possible with CSS variables.
This PR uses a CSS variable for input-border-thickness and checks the devices pixelRatio to either give it a 1.5px border or a 1px border.

Additionally the calculation for taking the border width into account was moved from $input-vertical-padding to the components themselves because sometime the border has to be removed twice (button) or only once (input).

## QA
- Open demo
- Look at any input element
- Verify, that the correct border amount has been substracted from the padding - on both high (>1.99dppx) and low pixel density screens.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
### Before
![Screenshot 2024-05-02 at 10 31 47](https://github.com/canonical/vanilla-framework/assets/68336485/12b67553-0078-45ed-a347-45909c385f8f)

### After
![Screenshot 2024-05-02 at 10 31 08](https://github.com/canonical/vanilla-framework/assets/68336485/cf0b8098-cfab-44c4-967c-4cae6da74c87)
